plutus-contract/test/Spec/Balancing.hs

```diff
@@ -165,6 +168,39 @@ balanceTxnNoExtraOutput =
 
     in checkPredicate "balancing doesn't create extra output" tracePred (void trace)
 
+balanceTxnCollateralTest :: TestName -> Int -> Integer -> TestTree
+balanceTxnCollateralTest name count lovelace =
+    let ee = someTokenValue "ee" 1
+        ff = someTokenValue "ff" 1
+        -- Make sure wallet 1 has only one utxo available.
+        options = defaultCheckOptions
+            & changeInitialWalletValue w1 (const $ Value.scale 1000 (ee <> ff) <> Ada.lovelaceValueOf 3_900_000)
+        vHash = Scripts.validatorHash someValidator
+
+        contract :: Contract () EmptySchema ContractError ()
+        contract = do
+            utxos <- Con.ownUtxos
+            let constraints1 =
+                    L.Constraints.mustPayToOtherScriptWithDatumInTx
+                        vHash
+                        unitDatum
+                        (Value.scale 100 ff <> Ada.lovelaceValueOf lovelace)
+                lookups = Tx.Constraints.unspentOutputs utxos
+                utx1 = either (error . show) id $ L.Constraints.mkTx @Void lookups constraints1
+            submitTxConfirmed utx1
+
+        trace = do
+            void $ Trace.activateContractWallet w1 contract
+            void $ Trace.waitNSlots 1
+
+    in checkPredicateOptions options name (assertValidatedTransactionCount count) trace
+
+balanceTxnCollateral :: TestTree
+balanceTxnCollateral = balanceTxnCollateralTest "can balance collateral with non-ada utxo" 1 1_200_000
+
+balanceTxnCollateralFail :: TestTree
+balanceTxnCollateralFail = balanceTxnCollateralTest "won't create return collateral with too little ada" 0 2_000_000
```

Comment:
Too little ada = not enough Ada per the costPerUtxoWord parameter?

Reply:
Good point. Should clarify in message that it's not proceeding due to minUtxo.